### PR TITLE
chore(cli): replace app init spinner in favor of progress tracker

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/cloudformation.go [hunk 1]
```diff
@@ -172,14 +172,38 @@ func (cf CloudFormation) errorEvents(stackName string) ([]string, error) {
...
 }
 
-func (cf CloudFormation) newRenderWorkloadInput(w progress.FileWriter, stack *cloudformation.Stack) *renderStackChangesInput {
-	in := &renderStackChangesInput{
+func (cf CloudFormation) newCreateChangeSetInput(w progress.FileWriter, stack *cloudformation.Stack) *executeAndRenderChangeSetInput {
+	in := &executeAndRenderChangeSetInput{
+		stackName:        stack.Name,
+		stackDescription: fmt.Sprintf("Creating the infrastructure for stack %s", stack.Name),
+	}
+	in.createChangeSet = func() (changeSetID string, err error) {
```
Comment (dannyrandall) [inline internal/pkg/deploy/cloudformation/cloudformation.go]: nit
```suggestion
	in.createChangeSet = func() (string, error) {
```

File: internal/pkg/deploy/cloudformation/cloudformation.go [hunk 2]
```diff
@@ -172,14 +172,38 @@ func (cf CloudFormation) errorEvents(stackName string) ([]string, error) {
...
+		var errAlreadyExists *cloudformation.ErrStackAlreadyExists
+		changeSetID, err = cf.cfnClient.Create(stack)
+		switch {
+		case err != nil && !errors.As(err, &errAlreadyExists):
+			spinner.Stop(log.Serrorf("%s\n", label))
+			return "", cf.handleStackError(stack.Name, err)
+		default:
+			spinner.Stop(log.Ssuccessf("%s\n", label))
+			return changeSetID, err
+		}
```
Comment (dannyrandall) [inline internal/pkg/deploy/cloudformation/cloudformation.go]: nit (i  `switch` but there's only one condition 🙄)
```suggestion
		if err != nil && !errors.As(err, &errAlreadyExists) {
		    spinner.Stop(log.Serrorf("%s\n", label))
			return "", cf.handleStackError(stack.Name, err)
		}
		
		spinner.Stop(log.Ssuccessf("%s\n", label))
		return changeSetID, err
```

File: internal/pkg/template/templates/app/app.yml [hunk 3]
```diff
@@ -153,6 +161,8 @@ Resources:
       Name: !Sub ${AppName}.${AppDomainName}
 
   AppDomainDelegationRecordSet:
+    Metadata:
+      'aws:copilot:description': 'Add NS records to delegate responsibility to the {{.Name}}.{{.Domain}} subdomain'
```
Comment (dannyrandall) [inline internal/pkg/template/templates/app/app.yml:165]:
```suggestion
      'aws:copilot:description': 'Add NS records to delegate responsibility to the {{.Name}}.{{.Domain}} hosted zone'
```

File: internal/pkg/deploy/cloudformation/app_test.go [hunk 4]
```diff
@@ -99,6 +100,7 @@ func TestCloudFormation_DeployApp(t *testing.T) {
 				cfnClient:   tc.mockStack(ctrl),
 				appStackSet: tc.mockStackSet(t, ctrl),
 				region:      tc.region,
+				console:     new(discardFile),
```
Comment (dannyrandall) [inline internal/pkg/deploy/cloudformation/app_test.go:103]: 😔

**Review by dannyrandall — APPROVED** [review]
looks great!😍 just a couple of nits you can ignore if you want😊

File: internal/pkg/deploy/cloudformation/cloudformation.go (hunk 2 above)
Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/cloudformation.go]: derp, thanks! fixed

File: internal/pkg/deploy/cloudformation/cloudformation.go [hunk 5]
```diff
@@ -219,11 +241,14 @@ func (cf CloudFormation) newRenderWorkloadInput(w progress.FileWriter, stack *cl
 	return in
 }
 
-func (cf CloudFormation) renderStackChanges(in *renderStackChangesInput) error {
+func (cf CloudFormation) executeAndRenderChangeSet(in *executeAndRenderChangeSetInput) error {
```
Comment (paragbhingre) [inline internal/pkg/deploy/cloudformation/cloudformation.go:244]:
```suggestion
func (cf CloudFormation) createAndRenderChangeSet(in *executeAndRenderChangeSetInput) error {
```

Is it okay if we keep the name of this method as `createAndRenderChangeSet` as `execute` sounds more of an execution of something and not creation of the stack. `createAndRenderChangeSet` name is also consistent with `createAndRenderEnvironment`. Not a major one so feel free to ignore if doesn't feel relevant.

**Review by paragbhingre — APPROVED** [review]
Yayyy progress tracker for app init 🎉

File: internal/pkg/deploy/cloudformation/cloudformation.go (hunk 5 above)
Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/cloudformation.go:244]: >Is it okay if we keep the name of this method as createAndRenderChangeSet as execute sounds more of an execution of something and not creation of the stack.

That's exactly right! we're executing the change set in this method that's why I renamed it. The `createChangeSet` function is a bit of misnomer I'll update in the follow-up PR, it actually either:
- creates and executes a 'Create' changeset
- or creates and executes an 'Update' changeset

**Pull request merged**